proposals/0000-Move-Platforms-to-Own-Packages.md

```diff
@@ -0,0 +1,273 @@
+   * `View`
+   * `VirtualizedList`
+   * `WebView` (DEPRECATED. [Moved to its own repo](https://github.com/react-native-community/react-native-webview))
+   * **APIs**
+   * `Alert`
+   * `Animated`
+   * `AppRegistry`
+   * `AppState`
+   * `AsyncStorage`
+   * `BackHandler`
+   * `CameraRoll`
+   * `Clipboard`
+   * `DeviceInfo`
+   * `Dimensions`
+   * `Easing`
+   * `findNodeHandle`
+   * `I18nManager`
+   * `InteractionManager`
+   * `Keyboard`
+   * `LayoutAnimation`
+   * `Linking`
+   * `NativeEventEmitter`
+   * `NetInfo`
+   * `PanResponder`
+   * `PixelRatio`
+   * `Settings`
+   * `Share`
+   * `StyleSheet`
+   * `Systrace`
+   * `TVEventHandler`
+   * `UIManager`
+   * `unstable_batchedUpdates`
+   * `Vibration`
+   * `YellowBox`
+   * **Plugins**
+   * `DeviceEventEmitter`
+   * `NativeAppEventEmitter`
+   * `NativeModules`
+   * `Platform`
+   * `processColor`
+   * `requireNativeComponent`
+   * `takeSnapshot`
+   * **Deprecated PropTypes**
+   * `ColorPropType`
+   * `EdgeInsetsPropType`
+   * `PointPropType`
+   * `ViewPropTypes`
+
+ * `react-native-ios`
+   * **Components**
+   * `DatePickerIOS`
+   * `InputAccessoryView`
+   * `MaskedViewIOS`
+   * `PickerIOS`
+   * `ProgressViewIOS`
+   * `SegmentedControlIOS`
+   * `SnapshotViewIOS`
+   * `TabBarIOS`
+   * **APIs**
+   * `ActionSheetIOS`
+   * `AlertIOS`
+   * `ImagePickerIOS`
+   * `PushNotificationIOS`
+   * `StatusBarIOS`
+   * `VibrationIOS`
+
+ * `react-native-android`
+   * **Components**
+   * `DrawerLayoutAndroid`
+   * `ProgressBarAndroid`
+   * `ToastAndroid`
+   * `ToolbarAndroid`
+   * `ViewPagerAndroid`
+   * **APIs**
+   * `DatePickerAndroid`
+   * `PermissionsAndroid`
+   * `TimePickerAndroid`
+
+
+### Hypothetical project overview (non-exhaustive)
+
+This is the *very basic* overview of what the GitHub repo could look like:
+
+```
+-| project root
+--📁 {.circleci, .github, bots}/
+--📁 flow/
+--📁 flow-github/
+--📁 packages/
+---📁 react-native/
+----📁 Libraries/ (All cross-platform JS)
+----📁 ReactCommon/
+----📁 scripts/
+-----📄 launchPackager.bat
+-----📄 launchPackager.command
+-----📄 packager.sh
+----📁 third-party/
+----📁 tools/ (contains bzl build-defs)
+---📁 react-native-ios/
+----📁 Libraries/
+----📁 React/
+----📁 scripts/
+-----📄 ios-configure-glog.sh
+-----📄 ios-install-third-party.sh
+-----📄 react-native-xcode.sh
+----📁 third-party-podspecs/
+---📁 react-native-android/
+----📁 gradle/
+----📁 keystores/
+----📁 Libraries/
+----📁 ReactAndroid/
+---📁 react-native-git-upgrade/
+--📁 RNTester/
+--📁 scripts/ (CI, sync, etc. scripts)
+--📄 package.json (for the Yarn workspace)
+--📄 {readme, contributing, CoC, other markdown docs}
+```
+
+I am not fully sure what the layout of the local-cli stuff will end up needing to be after this; with [this proposal](https://github.com/react-native-community/discussions-and-proposals/blob/master/proposals/0002-Extracted-Command-Line-Tools.md) the bulk of it will end up getting moved to another repo anyways.
```

Comment:
> OSS RN packages being published without being compiled

IMO I kinda like them shipping un-compiled. Especially with JSI allowing for different JS engines to be used with React Native, it would be useful for it to be easy to use different babel configs for different engines (imo). Biggest thing would be opting in to using native async/await where available for a cleaner error stack trace

Would be worth a discussion of its own for sure!

Reply:
There's a difference between not compiling parts of the code that are effectively js standards but not available in older environments, and publishing code with early stage proposals